src/Providers/Stripe/Customer/Service.php

```diff
@@ -26,15 +28,121 @@ public function create(Token $token, string $email, string $description): Custom
+        switch ($token->type) {
+            case StripeToken::PAYMENT_METHOD:
+                $params['payment_method'] = $token->value;
+
+                break;
+            case StripeToken::TOKEN:
+                $params['source'] = $token->value;
+
+                break;
+            default:
+                //
+        }
+
+        /** @var StripeCustomer $customer */
+        $response = StripeCustomer::create($params);
+
+        return new Customer($response->id, $response->email, $response->description);
+    }
+
+    /**
+     * Tries retrieve a customer using the Stripe SDK.
+     *
+     * @param Token $token
+     *
+     * @return Customer
+     */
+    public function find(Token $token): Customer
+    {
+        /** @var StripeCustomer $customer */
+        $response = StripeCustomer::retrieve($token->value);
+
+        return new Customer($response->id, $response->email, $response->description);
+    }
+
+    /**
+     * Tries add a card to a customer using the Stripe SDK.
+     *
+     * @param Customer $customer
+     * @param Card $card
+     *
+     * @return Card
+     */
+    public function addCard(Customer $customer, Card $card): Card
+    {
+        /** @var PaymentMethod $paymentMethod */
+        $paymentMethod = PaymentMethod::retrieve($card->uid);
+
+        $response = $paymentMethod->attach(['customer' => $customer->uid]);
+
+        return new Card(
+            $response->id,
+            $response->card->last4,
+            $response->card->brand,
+            $response->card->funding,
+            $response->card->exp_month,
+            $response->card->exp_year
+        );
+    }
+
+    /**
+     * Tries remove a source API card from a customer using the Stripe SDK.
+     *
+     * @param Customer $customer
+     * @param Card $card
+     *
+     * @return void
+     */
+    public function deleteCard(Customer $customer, Card $card): void
+    {
+        StripeCustomer::deleteSource($customer->uid, $card->uid);
+    }
+
+    /**
+     * Tries get customer's cards using the Stripe SDK.
+     *
+     * @param Customer $customer
+     *
+     * @return Card[]
+     */
+    public function getCards(Customer $customer): array
+    {
+        $cards = [];
+
+        $paymentMethods = PaymentMethod::all([
+            "customer" => $customer->uid,
+            "type" => "card",
+        ]);
+
+        foreach ($paymentMethods->data as $paymentMethod) {
+            $cards[] = new Card(
+                $paymentMethod->id,
+                $paymentMethod->card->last4,
+                $paymentMethod->card->brand,
+                $paymentMethod->card->funding,
+                $paymentMethod->card->exp_month,
+                $paymentMethod->card->exp_year
+            );
         }
 
         /** @var StripeCustomer $customer */
-        $customer = StripeCustomer::create($params);
+        $customer = StripeCustomer::retrieve($customer->uid);
+
+        if (property_exists($customer, 'cards')) {
```

Comment:
actually I'm going to change that because looking at the documentation we should use Stripe\Customer::allSources(...) instead. As the customer object seem to have been modified in the new version

See : https://stripe.com/docs/api/cards/list